### PR TITLE
Node stats API

### PR DESCRIPTION
**Pull request by dhyaniarun1993**
Closes: #68

### CODE REVIEW

File: store/mongo/mongo.go [hunk 1]
```diff
@@ -272,6 +272,46 @@ func (s *mongoStore) AggregateByNetworkType(ctx context.Context) ([]*models.Aggr
...
+type syncAggregateData struct {
+	IsSynced bool `json:"_id" bson:"_id"`
+	Count    int  `json:"count" bson:"count"`
+}
+
+func (s *mongoStore) AggregateBySyncStatus(ctx context.Context) ([]*models.AggregateData, error) {
+	query := mongo.Pipeline{
+		bson.D{
+			{Key: "$group", Value: bson.D{
+				{Key: "_id", Value: "$issynced"},
```
Comment (sadiq1971) [inline store/mongo/mongo.go:284]: Is it `issynced` or `is_synced` ?

File: store/mongo/mongo.go [hunk 2]
```diff
@@ -272,6 +272,46 @@ func (s *mongoStore) AggregateByNetworkType(ctx context.Context) ([]*models.Aggr
...
+				{Key: "count", Value: bson.D{{Key: "$sum", Value: 1}}},
+			}},
+		},
+	}
+	cursor, err := s.coll.Aggregate(ctx, query)
```
Comment (sadiq1971) [inline store/mongo/mongo.go:289]: Why we are not using `s.coll.Count...` here? We can simply count the synced nodes and total nodes!

File: store/mongo/mongo.go (hunk 1 above)
Comment (dhyaniarun1993) [reply store/mongo/mongo.go:284]: The peer model has it as issynced.
We have json tags but in mongoDB bson tags are used and we haven't defined them.
I guess we need to fix the field names

File: store/mongo/mongo.go (hunk 2 above)
Comment (dhyaniarun1993) [reply store/mongo/mongo.go:289]: So if we use count, we have to perform two queries one to get count of all nodes and another to get count of synced nodes. 
And if we do that, there can be the case where nodes are added between the two queries. So to get full consistent view of node, I went with single aggregation query to get both the information.

File: store/mongo/mongo.go (hunk 1 above)
Comment (sadiq1971) [reply store/mongo/mongo.go:284]: yes. need to define bson tag for all fields. I thought undefined field will be tagged same as json

File: store/mongo/mongo.go (hunk 2 above)
Comment (sadiq1971) [reply store/mongo/mongo.go:289]: So, Is the cursor returning only two result? I mean the len of result is 2?

Comment (dhyaniarun1993) [reply store/mongo/mongo.go:289]: As of now two fields.But it could potentially returns three fields `true`, `false` and `null`.

**Review by sadiq1971 — APPROVED** [review]

**Pull request merged**